### ISSUE TITLE
Possible bug with .ge gradient flow?

### ISSUE THREAD

**Issue by Abhishaike** [pytorch/pytorch#7322]
I may have a fundamental misunderstanding how how PyTorch works, but whenever I try to apply a torch.ge() function in the middle of the forward() definition, gradients do not flow to the layers prior to that function. 

Here is a reproducible code example: 
```
import torch
import torch.nn as nn
from torch.autograd import Variable
import torchvision
import torch.nn.functional as F
import torch.optim as optim
import torchvision.transforms as transforms

class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.layer1_0 = self.Conv_Module(3, 6, 1, 3, 1)
        self.layer1_1 = self.Conv_Module(1, 6, 4, 3, 1)
        self.Dense = self.Classification(384)

    def Conv_Module(self, in_filters, out_filters, stride, kernel_size, padding):
        return nn.Sequential(
            nn.Conv2d(in_channels=in_filters, out_channels=out_filters, kernel_size=kernel_size, stride=1, padding=padding),
            nn.ReLU(),
            nn.Conv2d(in_channels=out_filters, out_channels=out_filters, kernel_size=kernel_size, stride=stride, padding=padding),
            nn.ReLU())

    def Classification(self, num_in):
        return nn.Sequential(
            nn.Linear(num_in, 10))

    def forward(self, original_img):
        first_conv = self.layer1_0(original_img)
        attention_map = first_conv.mean(1).unsqueeze(dim = 1)
        inverted_attention = -attention_map
        output_mask = torch.ge(attention_map, inverted_attention).float()
        second_conv = self.layer1_1(output_mask)
        Classifier = self.Dense(second_conv.view(-1, 384))
        return F.log_softmax(Classifier, dim=1)


transform = transforms.Compose([
    transforms.ToTensor(),
    transforms.Normalize((.5, .5, .5), (.5, .5, .5))])
trainset = torchvision.datasets.CIFAR10(root='./data', train=True,
                                        download=True, transform=transform)
train_loader = torch.utils.data.DataLoader(trainset, batch_size=128,
                                           shuffle=True, num_workers=1)

model = Net()
optimizer = optim.Adam(model.parameters(), lr=1.5)
model.train()

for batch_idx, (data, target) in enumerate(train_loader):
    model.train()
    data, target = Variable(data), Variable(target)
    optimizer.zero_grad()
    output = model(data)
    loss = F.nll_loss(output, target)
    loss.backward()
    optimizer.step()
    print("Weights before .ge:" , torch.sum(model.layer1_0[0].weight))
    print("Weights after .ge:" ,torch.sum(model.layer1_1[0].weight))
    print("*********")
```

The print statements I have at the bottom show that the weights change for the layers after torch.ge(), but do not change before it. Here's what the output is:

```
Weights before .ge: tensor(-1.2797)
Weights after .ge: tensor(10.8658)
*********
Weights before .ge: tensor(-1.2797)
Weights after .ge: tensor(-24.6382)
*********
Weights before .ge: tensor(-1.2797)
Weights after .ge: tensor(-62.4911)
*********
Weights before .ge: tensor(-1.2797)
Weights after .ge: tensor(-93.4978)
*********
Weights before .ge: tensor(-1.2797)
Weights after .ge: tensor(-119.7056)
*********
Weights before .ge: tensor(-1.2797)
Weights after .ge: tensor(-142.2933)
*********
Weights before .ge: tensor(-1.2797)
Weights after .ge: tensor(-162.0192)
```
Please try to provide a minimal example to repro the bug.
Error messages and stack traces are also helpful.

## System Info
Collecting environment information...
PyTorch version: 0.4.0
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.13.3
GCC version: Could not collect
CMake version: version 3.10.3

Python version: 3.5
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip] numpy (1.8.0rc1)
[conda] Could not collect

**Comment by SsnL** [comment]
“greater than or equal” has zero gradient almost everywhere, and nondifferentiable at other points. It’s not a bug.

**Comment by Alex6D** [comment]
Have you solved this issue? I run into the same needs, the gradient seems to be interrupted. Thanks
